### DISCUSSION TITLE
[Accordion] Do we really need `padding` prop?

### DISCUSSION THREAD

**Discussion by GomezIvann** [dxc-technology/halstack-react#1266] [Ideas] [answered]
Since we recently launched the new Layout components, do we want to keep this kind of props or should we start suggesting using, in this case, the Inset component for a more semantic layout?

```
      <DxcAccordion label="Personal information">
        <DxcInset space="2rem">
          <DxcTextInput label="Name" />
          <DxcTextInput label="Surname" />
        </DxcInset>
      </DxcAccordion>
```

**Comment by aidamag** [comment]
Good point, I also think we should use the DxcInset. We should check the rest of the components to apply this change too.

**Comment by raquelarrojo** [comment]
I totally agree. Now that we have layout components, it is a good point to remove `padding` in Accordion. In the future I will also remove `margin` for all the components.

**Comment by kukoc** [comment]
As much as I love using the Layout components, (they make easier the life of a CSS ignorant like me), removing the padding prop from the Accordion is a breaking change, isn't it? We must be careful with these changes. Moreover, if we consider applying this change across all DS components. Let's think about it.

**Reply by GomezIvann** [reply]
I think you have a point here Aitor, we must be careful with what we change, since the `margin` prop, for example, is in most of the components. That's why I only mention `padding`, which is also a breaking change but is "only" in the Accordion, Accordion Group, Box, Dialog, Footer, and Header components. 
The problem here is that we want them (the users) to use our Layout both outside and inside of our components. Props like `padding` make the usage of Inset worthless and more extensive. Who wants to import and use another component when you already have a prop? We can't expect them to use the Layout components if they already have other solutions, proposed by as, which are shorter and are currently legit too...

**Comment by jsuarezgonz** [comment]
I agree with using the DxcInset component within our components and removing the prop padding; I would go further and also replace the prop margin by using one of our layout components.

Even if it´s a breaking change I think it´s something that we should use to give more consistency and make use of the components in the same way that we want the users to use them. I know it´s a breaking change but as we always said, I see the Design System as something alive and evolving.

As well as I think that the use of DxcInset within the components is a good idea, I think that we should use our Typography components within the rest of the components. For example, use the DxcTypography component for component labels.

**Comment by GomezIvann** [comment, marked as answer]
As we are all more or less in agreement, we can start removing the `padding` prop from the DS components and discuss if we want to preserve other related ones such as `margin`.

**Reply by marcialfps** [reply]
Yes! We can add it to the backlog. ✌🏼
